plugin/trino-lancedb/src/test/java/io/trino/plugin/lance/TestLanceDatasetPageSource.java

```diff
@@ -0,0 +1,43 @@
+package io.trino.plugin.lance;
+
+import com.google.common.io.Resources;
+import io.trino.plugin.lance.internal.LanceReader;
+import io.trino.spi.Page;
+import io.trino.spi.connector.ConnectorTableHandle;
+import io.trino.spi.connector.SchemaTableName;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.net.URL;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestLanceDatasetPageSource
+{
+    private static final SchemaTableName TEST_TABLE_1 = new SchemaTableName("default", "test_table1");
+
+    private LanceMetadata metadata;
+
+    @BeforeEach
+    public void setUp()
+            throws Exception
+    {
+        URL lanceDbURL = Resources.getResource(LanceReader.class, "/example_db");
+        assertThat(lanceDbURL)
+                .describedAs("example db is null")
+                .isNotNull();
+        LanceConfig lanceConfig = new LanceConfig().setLanceDbUri(lanceDbURL.toString());
+        LanceReader lanceReader = new LanceReader(lanceConfig);
+        metadata = new LanceMetadata(lanceReader, lanceConfig);
+    }
+
+    @Test
+    public void testTableScan()
```

Comment:
this test failed with the following command

```
mvn install -Dair.check.skip-all -pl plugin/trino-lancedb
mvn test -Dair.check.skip-all -pl plugin/trino-lancedb -Dtest=TestLanceDatasetPageSource*
```

the failure is relatively hard to debug but i can't be sure what's wrong. we can probably first walk through the test case and see what's wrong first. and next we can chat about what's the proper way to deal with it. 

problem with running in intellij is that it attaches self during java debugging and thus making a AttachMain <--> JUnitStarter process dependency loop. and i cant kill them.

running with mvn is good but i cant debug what's wrong.

Reply:
CC @eddyxu